pkg/cincinnati/cincinnati.go

```diff
@@ -331,33 +391,36 @@ func GetChannels(ctx context.Context, c Client, channel string) (map[string]stru
 	return channels, nil
 }
 
-// GetVersion will return all OCP/OKD versions in a specified channel fetches the current and requested (if applicable) update payload from the specified
-// upstream Cincinnati stack given the current version and channel
-func GetVersions(ctx context.Context, c Client, channel string) ([]semver.Version, error) {
+// GetVersions will returns all update payloads from the specified
+// upstream Cincinnati stack given architecture and channel.
+func GetVersions(ctx context.Context, c Client, arch, channel string) ([]semver.Version, error) {
 	// Prepare parametrized cincinnati query.
-	c.SetQueryParams("", channel, "")
+	c.SetQueryParams(arch, channel, "")
 
 	graph, err := getGraphData(ctx, c)
 	if err != nil {
-		return nil, fmt.Errorf("error getting graph data for channel %s", channel)
+		return nil, &Error{
+			Reason:  "APIRequestError",
+			Message: fmt.Sprintf("channel %q: %v", channel, err),
+			cause:   err,
```

Comment:
`Cause`?

Reply:
Suggesting it be exported or asking why it is part of the spec?

Reply:
I'm just a bit curious why it is hidden compared to Reason and Message. Not a big deal.

Reply:
Gotcha, I think it's a fair question, I was just making sure I understood. This structured error was brought over when we initially modeled this after the CVO operator. The `cause` field is actually not used currently, but I agree it could be exported in the event that errors need to be unwrapped outside of the `cincinnati` package.